src/main/java/edu/ucar/unidata/rosetta/service/DataManagerImpl.java

```diff
@@ -0,0 +1,283 @@
+
+    /**
+     * Looks up and retrieves a Data object using the given id.
+     *
+     * @param id    The id of the Data object.
+     * @return      The Data object corresponding to the given id.
+     */
+    @Override
+    public Data lookupById(String id) {
+        return dataDao.lookupById(id);
+    }
+
+    /**
+     * Persists the information in the given data object.
+     *
+     * @param data  The Data object to persist.
+     */
+    @Override
+    public void persistData(Data data, HttpServletRequest request) {
+        data.setId(createUniqueDataId(request)); // Create a unqiue ID for this object.
+        if (data.getPlatform() != null) {
+            String community = resourceManager.getCommunity(data.getPlatform());
+            data.setCommunity(community);
+        }
+        dataDao.persistData(data);
+    }
+
+    /**
+     * Updated the persisted information corresponding to the given data object.
+     *
+     * @param data  The data object to update.
+     */
+    @Override
+    public void updateData(Data data) {
+        dataDao.updatePersistedData(data);
+    }
+
+    /**
+     * Deletes the persisted data object information.
+     *
+     * @param id    The id of the Data object to delete.
+     */
+    @Override
+    public void deleteData(String id) {
+        dataDao.deletePersistedData(id);
+    }
+
+    /**
+     * Retrieves the name of the directory used for storing uploaded files.
+     *
+     * @return  The name of the directory used for storing uploaded files.
+     */
+    @Override
+    public String getUploadDir() {
+        return propertiesDao.lookupUploadDirectory();
+    }
+
+    /**
+     * Combines non-null user-provided entries with persisted entries in a single Data object.
+     * A new Data is created from the persisted data and is populated with the non-null values
+     * of the user-provided information using reflection.
+     *
+     * @param id    The id of the persisted data associated with the provided Data object.
+     * @param data  The Data object containing the user-provided data.
+     * @return  A data object that contains the new user-provided data and the persisted data.
+     * @throws RosettaDataException  If unable to populate the Data object by reflection.
+     */
+    public Data populateDataObjectWithInput(String id, Data data) throws RosettaDataException {
+        // Data persisted in the database.
+        Data persistedData = lookupById(id);
+
+        try {
+            for (Method method : data.getClass().getDeclaredMethods()) {
+                if (Modifier.isPublic(method.getModifiers())
+                        && method.getParameterTypes().length == 0
+                        && method.getReturnType() != void.class
+                        && (method.getName().startsWith("get"))
+                        ) {
+                    Object value = method.invoke(data);
+                    if (value != null) {
+                        if (value instanceof String) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)